source/val/construct.cpp

```diff
@@ -13,10 +13,13 @@
 // limitations under the License.
 
 #include "val/construct.h"
+#include "val/function.h"
 
 #include <cassert>
 #include <cstddef>
+#include <unordered_set>
 
+#include <iostream>
```

Comment:
I don't think this include is needed.  If it is, group it with the group above it.